indico/modules/rb_new/client/js/modules/bookRoom/BookingSuggestion.jsx

```diff
@@ -0,0 +1,130 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Grid, Icon, Label, Message, Popup} from 'semantic-ui-react';
+import {PluralTranslate, Translate, Singular, Param, Plural} from 'indico/react/i18n';
+import {Slot, IndicoPropTypes} from 'indico/react/util';
+
+import Room from '../../components/Room';
+
+import './BookingSuggestion.module.scss';
+
+
+function SuggestionOption({onClick, overrides, icon, text, tooltip}) {
+    const message = (
+        <Message styleName="suggestion-text" size="mini"
+                 onClick={onClick.bind(undefined, overrides)}
+                 warning compact>
+            <Message.Header>
+                <Icon name={icon} /> {text}
+            </Message.Header>
+        </Message>
+    );
+    return (
+        <Popup trigger={message}
+               content={tooltip}
+               position="right center" />
+    );
+}
+
+SuggestionOption.propTypes = {
+    onClick: PropTypes.func.isRequired,
+    overrides: PropTypes.object,
+    icon: PropTypes.string.isRequired,
+    text: IndicoPropTypes.i18n.isRequired,
+    tooltip: IndicoPropTypes.i18n.isRequired
+};
+
+SuggestionOption.defaultProps = {
+    overrides: null
+};
+
+/**
+ * Renders a "suggested room" containing one or more alternative options
+ * e.g. "10 minutes later", "Skip one day"...
+ */
+export default class BookingSuggestion extends React.PureComponent {
+    static propTypes = {
+        /** Room information */
+        room: PropTypes.object.isRequired,
+        /** Suggestions as {time, duration, skip} */
+        suggestions: PropTypes.object.isRequired,
+        /** Will be called when a suggestion is chosen (clicked on) */
+        onClick: PropTypes.func.isRequired
+    };
+
+    renderSuggestionText(room, {time, duration, skip}) {
+        const {onClick} = this.props;
+        return (
+            <>
+                {time && (
+                    <SuggestionOption onClick={onClick}
+                                      overrides={{time}}
+                                      icon="clock"
+                                      text={(
```

Comment:
what about using children for this instead of a normal prop?

Reply:
Thought of it, but then there's the "tooltip" field as well. We could also use `Slot`s, but passing as an arg doesn't seem to be an uncommon pattern in React.

Reply:
Just thought because it's quite big. But since the code to generate this message will probably end up in a separate function to fix the i18n I guess that point becomes moot anyway